realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -113,6 +113,10 @@ public DynamicRealmObject(RealmModel obj) {
      */
     public boolean getBoolean(String fieldName) {
         long columnIndex = proxyState.getRow$realm().getColumnIndex(fieldName);
+        RealmFieldType columnType = proxyState.getRow$realm().getColumnType(columnIndex);
```

Comment:
Yes, if we are willing to accept more grammatical errors, i.e as the `is not a(n) X, but a(n) Y`.
It will make the code more maintainable though. Thoughts?